docs/data/ingest-library/BufferedStorageBackend.mdx

```diff
@@ -0,0 +1,159 @@
+---
+title: BufferedStorageBackend
+sidebar_position: 0
+---
+
+# BufferedStorageBackend: Retrieving Ledger Metadata from the Data Lake
+
+The [BufferedStorageBackend](https://github.com/stellar/go/blob/bd7c1ad981593adf37b6bb6b91ef1d3f9afb1f00/ingest/ledgerbackend/buffered_storage_backend.go) is a component of the Stellar [ingest](https://github.com/stellar/go/tree/bd7c1ad981593adf37b6bb6b91ef1d3f9afb1f00/ingest) library that retrieves ledger metadata from a data lake generated by Stellar [Galexie](https://developers.stellar.org/docs/data/galexie). While Stellar Galexie currently supports only [GCS](https://developers.stellar.org/docs/data/galexie/admin_guide/configuring) as the data lake, the `BufferedStorageBackend` can work with any datastore that implements the [datastore interface](https://github.com/stellar/go/blob/master/support/datastore/datastore.go). The `BufferedStorageBackend` returns ledger metadata in [XDR](https://developers.stellar.org/docs/learn/encyclopedia/data-format/xdr) format.
+
+## Key Features
+
+- **Parallel Downloads**: Downloads multiple ledgers concurrently and buffers them in memory for fast access. This is particularly useful for fetching large historical ledger ranges.
+- **Schema-Aware**: Reads multi-ledger files based on the datastore schema, extracting one ledger at a time.
+- **Automatic Retries**: Handles request failures by retrying failed requests.
+- **XDR Output**: Returns ledger metadata in XDR format, enabling easy integration with other packages in the ingest library (e.g., [processors](https://github.com/stellar/go/tree/bd7c1ad981593adf37b6bb6b91ef1d3f9afb1f00/ingest/processors)).
+
+## Prerequisites
+
+- A data lake created by Stellar Galexie. For instructions on how to create a data lake, refer to the [Stellar Galexie administrator guide](https://developers.stellar.org/docs/data/galexie/admin_guide).
```

Comment:
```suggestion
- Access to a data lake populated by Galexie. For purposes of the example code, the data lake should be in the form of a GCS bucket. For instructions on how to create a data lake, refer to the [Stellar Galexie administrator guide](https://developers.stellar.org/docs/data/galexie/admin_guide). 
```
It seems worthwhile to state relation of 'data lake' to GCS bucket, otherwise reader may not be sure what it refers to.